zboxcore/sdk/sdk.go

```diff
@@ -609,34 +610,74 @@ func (v *Validator) ConvertToValidationNode() *blockchain.ValidationNode {
 	}
 }
 
-func GetBlobbers() (bs []*Blobber, err error) {
-	if !sdkInitialized {
-		return nil, sdkNotInitialized
+func getBlobbersInternal(active bool, limit, offset int) (bs []*Blobber, err error) {
+	type nodes struct {
+		Nodes []*Blobber
 	}
 
-	var b []byte
-	b, err = zboxutil.MakeSCRestAPICall(STORAGE_SCADDRESS, "/getblobbers?active=true", nil,
-		nil)
+	url := fmt.Sprintf("/getblobbers?active=%s&limit=%d&offset=%d",
+		strconv.FormatBool(active),
+		limit,
+		offset,
+	)
+	b, err := zboxutil.MakeSCRestAPICall(STORAGE_SCADDRESS, url, nil, nil)
+	var wrap nodes
 	if err != nil {
 		return nil, errors.Wrap(err, "error requesting blobbers:")
 	}
 	if len(b) == 0 {
 		return nil, errors.New("", "empty response")
 	}
 
-	type nodes struct {
-		Nodes []*Blobber
-	}
-
-	var wrap nodes
-
 	if err = json.Unmarshal(b, &wrap); err != nil {
 		return nil, errors.Wrap(err, "error decoding response:")
 	}
 
 	return wrap.Nodes, nil
 }
 
+type GetBlobbersOptions struct {
+	Active bool // set this to true to get only active blobber
+}
+
+func GetBlobbers(options ...GetBlobbersOptions) (bs []*Blobber, err error) {
+	if !sdkInitialized {
+		return nil, sdkNotInitialized
+	}
+
+	var active bool
+	if len(options) > 0 {
+		for _, option := range options {
+			active = option.Active
```

Comment:
Use a single bool optional param should be enough if we only have one option, eg, `GetBlobbers(activeOnly ...bool) `. Otherwise, use [optional function](https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis) to make it extendable.

Reply:
updated to this signature: `GetBlobbers(active ...bool)`